lib/chef/knife/bootstrap.rb

```diff
@@ -836,6 +838,7 @@ def ssh_opts
         return opts if connection_protocol == "winrm"
         opts[:non_interactive] = true # Prevent password prompts from underlying net/ssh
         opts[:forward_agent] = (config_value(:ssh_forward_agent) === true)
+        opts[:connection_timeout] = config_value(:session_timeout)&.to_i || 60
```

Comment:
Why the `&`?

Reply:
@btm Actually if config_value(:session_timeout) is nil `to_i` return `0` 
so 
`config_value(:session_timeout).to_i || 60` evaluates to `0` 
To avoid this 
`&` used to first check nil and return nil for for expression `config_value(:session_timeout)&.to_i`
so 
`config_value(:session_timeout)&.to_i || 60` evaluates to 60